testutil/testpolylog/event.go

```diff
@@ -0,0 +1,166 @@
+package testpolylog
+
+import (
+	"bytes"
+	"fmt"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+
+	"github.com/pokt-network/poktroll/pkg/polylog"
+)
+
+type (
+	// NewLoggerAndOutputFn is called in the test helper to create a new logger
+	// configured with the given level and options. It returns the logger and
+	// the buffer to which the logger writes. It is useful for decoupling test
+	// helpers from a specific logger implementation.
+	NewLoggerAndOutputFn func(
+		*testing.T,
+		polylog.Level,
+		...polylog.LoggerOption,
+	) (polylog.Logger, *bytes.Buffer)
+
+	// NewEventWithLevelFn is called in the test helper to create a new event
+	// at the given level from the given logger. It is useful for decoupling
+	// test helpers from a specific logger implementation.
+	NewEventWithLevelFn func(
+		*testing.T,
+		polylog.Logger,
+		polylog.Level,
+	) polylog.Event
+)
+
+// FnMethodSpy is a mock which implements a #Fn() method that is intended to be
```

Comment:
What if we rename `Fn` to `SpyOnEvent`?

Reply:
My rationale behind `FnMethodSpy` is that:
1. The object is a spy by definition of how we intend on using it.
2. Because the spy object must encapsulate the function we're spying on, it's becomes method which subsequently needs a name.
3. As we only need a single spy function, it seemed reasonable to name the method `Fn`.

Reply:
> As we only need a single spy function

I agree, but this seems like an implicit requirement. If we rename it to `FnMethodSpy`, the code is self documenting without needing to understand it as deeply.